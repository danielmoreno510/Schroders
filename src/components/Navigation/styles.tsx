import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
  },
  headerLeftContainer: {
    marginLeft: 12,
  },
  badge: {
    marginBottom: -12,
    marginLeft: 14,
    height: 22,
    width: 22,
    zIndex: 1,
  },
  badgeText: {
    color: 'white',
  },
});
