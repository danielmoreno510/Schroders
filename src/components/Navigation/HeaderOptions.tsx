import React from 'react';
import {TouchableOpacity} from 'react-native';
import {CloseIcon, HamburgerIcon, NativeBaseProvider} from 'native-base';

import styles from './styles';
import {primaryColor} from '../../constants';

const HeaderOptions: React.FC<HeaderOptionsProps> = ({
  changeScreen,
  isMenu,
}) => (
  <NativeBaseProvider>
    <TouchableOpacity onPress={changeScreen} style={styles.headerLeftContainer}>
      {isMenu ? (
        <CloseIcon size="5" color={primaryColor} />
      ) : (
        <HamburgerIcon size="6" color={primaryColor} />
      )}
    </TouchableOpacity>
  </NativeBaseProvider>
);

export default HeaderOptions;
