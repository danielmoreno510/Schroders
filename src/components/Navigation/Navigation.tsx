import React from 'react';
import {SafeAreaView} from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

import HeaderOptions from './HeaderOptions';
import Home from '../../screens/Home';
import Menu from '../../screens/Menu';
import CommitList from '../../screens/CommitList';
import RepositoryDetails from '../../screens/CommitDetails';
import About from '../../screens/About';
import styles from './styles';
import {primaryColor} from '../../constants';

const Stack = createStackNavigator();
const screenOptions = {
  headerStyle: {shadowColor: primaryColor},
  headerTintColor: primaryColor,
};

const Navigation: React.FC = () => {
  const headerClose = ({navigation}: Navigation) => ({
    headerLeft: () => (
      <HeaderOptions
        changeScreen={() => navigation.navigate('Home')}
        isMenu={true}
      />
    ),
  });

  const headerMenu = ({navigation}: any) => ({
    headerLeft: () => (
      <HeaderOptions
        changeScreen={() => navigation.navigate('Menu')}
        isMenu={false}
      />
    ),
  });

  return (
    <NavigationContainer>
      <SafeAreaView style={styles.container}>
        <Stack.Navigator headerMode="screen" screenOptions={screenOptions}>
          <Stack.Screen name="Home" component={Home} options={headerMenu} />
          <Stack.Screen
            name="Commit List"
            component={CommitList}
            options={headerClose}
          />
          <Stack.Screen name="Details" component={RepositoryDetails} />
          <Stack.Screen
            name="Menu"
            component={Menu}
            options={({navigation}) => ({
              ...headerClose({navigation}),
              gestureDirection: 'horizontal-inverted',
            })}
          />
          <Stack.Screen name="About" component={About} options={headerClose} />
        </Stack.Navigator>
      </SafeAreaView>
    </NavigationContainer>
  );
};

export default Navigation;
