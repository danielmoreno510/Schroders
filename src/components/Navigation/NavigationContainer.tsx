import Navigation from './Navigation';
import {StoreService} from '../../redux/StoreService';

const NavigationContainer = StoreService.connect()(Navigation);

export default NavigationContainer;
