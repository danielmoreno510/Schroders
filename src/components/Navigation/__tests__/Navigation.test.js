import React from 'react';
import TestRenderer from 'react-test-renderer';

import Navigation from '../Navigation';

describe('Navigation component', () => {
  describe('WHEN the component is rendered', () => {
    const testRenderer = TestRenderer.create(<Navigation />);
    const testInstance = testRenderer.root;

    testInstance
      .findByProps({testID: 'Home'})
      .props.options(jest.fn())
      .headerLeft();
    testInstance
      .findByProps({testID: 'Menu'})
      .props.options(jest.fn())
      .headerLeft();

    it('THEN should display a regular Navigation', () => {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
  });
});
