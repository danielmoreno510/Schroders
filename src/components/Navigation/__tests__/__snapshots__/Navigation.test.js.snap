// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component WHEN the component is rendered THEN should display a regular Navigation 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    component={[Function]}
    name="Home"
    options={[Function]}
    testID="Home"
  />
  <View
    component={[Function]}
    name="Commit List"
    options={[Function]}
    testID="Commit List"
  />
  <View
    component={[Function]}
    name="Details"
    testID="Details"
  />
  <View
    component={[Function]}
    name="Menu"
    options={[Function]}
    testID="Menu"
  />
  <View
    component={[Function]}
    name="About"
    options={[Function]}
    testID="About"
  />
</RCTSafeAreaView>
`;
